### PR TITLE
Normalize Arizona arcgis vaccine data (fixes #59)

### LINKED ISSUE #59: normalize arcgis from Arizona
[learn our pipeline: normalize](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Runner-pipeline-stages#normalize)

Normalize existing [`.ndjson`](http://ndjson.org/) records into [the Vaccinate The States data schema](https://github.com/CAVaccineInventory/vaccine-feed-ingest/blob/main/vaccine_feed_ingest/schema/schema.py).

Read all files in the directory passed as the second argument (`sys.argv[2]`), transform each record into the normalized schema (filling in as much detail as possible), then output them to new files in the directory passed as the first argument (`sys.argv[1]`).

Check the wiki to learn more about the purpose of the normalize stage and how to get set up for development!

### Tips

1. Fetch and parse data for this site before you start developing:
    ```sh
    poetry run vaccine-feed-ingest fetch az/arcgis
    poetry run vaccine-feed-ingest parse az/arcgis
    ```

1. While working on your code, run it at any point:
    ```sh
    poetry run vaccine-feed-ingest normalize az/arcgis
    ```

### Example
[Parsed files for `ak/arcgis`](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/parsed) are converted to [normalized files](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/normalized)

### PR DESCRIPTION
**Pull request by not-an-aardvark**
Partially adapted from the Alaska normalizer.

The main complexity is parsing site hours, which are in a very wide variety of time formats; it seems like a value is produced for almost all of them. I wasn't sure how deeply I should go into handling special cases versus keeping the normalizer simple; currently it's able to parse almost all of the times, but is fairly complicated and might not be robust. (Some sites have the hours listed for particular days as the string "BY APPOINTMENT ONLY" or "7:00 AM-5:00PM for second dose vaccine ONLY"; these are theoretically parseable, but it's not clear how they fit with the schema. The time parsing just falls back to `None` if it can't figure out what's going on.)

I checked for correctness by running `poetry run vaccine-feed-ingest normalize az/arcgis`, ensuring it ran to completion, and quickly skimming the output file to look for obvious errors. I also cross-checked a few of the normalized "hours" values to make sure they looked accurate, but I didn't thoroughly double-check the hours for all the sites.

Fixes #59

### CODE REVIEW

File: vaccine_feed_ingest/runners/az/arcgis/normalize.py [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+    return {
+        None: None,
+        "Yes": ["en", "es"],
+        "No": ["en"]
+    }[site["attributes"]["spanish_staff_y_n"]]
+
+
+def _get_opening_dates(site: dict) -> Optional[List[schema.OpenDate]]:
+    if site["attributes"]["begindate"] is None or site["attributes"]["enddate"] is None:
+        return None
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/az/arcgis/normalize.py:80]: It is okay to add _just_ an `opens` date or _just_ a `closes` date in the normalized schema (we commonly see sites which opened in the past - so might not have a date recorded, and sites which don't have a close date yet assigned but we don't want to miss later).

Could we loosen this restriction?

```suggestion
    if site["attributes"]["begindate"] is None and site["attributes"]["enddate"] is None:
        return None
```

File: vaccine_feed_ingest/runners/az/arcgis/normalize.py [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+
+    raise ValueError(human_readable_time)
+
+
+def _normalize_time(human_readable_time: str) -> str:
+    hour, minute = _parse_time(human_readable_time)
+    return str(hour % 24).rjust(2, '0') + ':' + str(minute).rjust(2, '0')
+
+
+def _normalize_hours(human_readable_hours: Optional[str], day: str) -> List[schema.OpenHour]:
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/az/arcgis/normalize.py:117]: Very impressive hour-parsing, thank you.

**Review by eliblock — COMMENTED** [review]
Great work - the hour-parsing logic you built is impressive across a very-not-standardized dataset.

Two suggestions - one around phone numbers, one around open/close days, but then we should be able to get this data on the map!

In order to pass the linter, you'll want to:
* `poetry run black .` - this will reformat your code
* `poetry run flake8 vaccine_feed_ingest/runners/az/arcgis/normalize.py` - this will show the same error the linter shows (but oftentimes running `black` fixes `flake8` errors for you)

File: vaccine_feed_ingest/runners/az/arcgis/normalize.py [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+    arcgis = "128ead309d754558ad81bccd99188dc9"
+    layer = 0
+
+    return f"{runner}:{site}:{arcgis}_{layer}:{data_id}"
+
+
+def _get_contacts(site: dict) -> Optional[List[schema.Contact]]:
+    contacts = []
+    if site["attributes"]["prereg_phone"]:
+        sourcePhone = re.sub("[^0-9]", "", site["attributes"]["prereg_phone"])
```
[the remark this answers is not in the document]
Comment (not-an-aardvark) [reply vaccine_feed_ingest/runners/az/arcgis/normalize.py]: Done.

File: vaccine_feed_ingest/runners/az/arcgis/normalize.py (hunk 1 above)
Comment (not-an-aardvark) [reply vaccine_feed_ingest/runners/az/arcgis/normalize.py]: Updated to allow having only one of `opens` and `closes`.

**Review by eliblock — APPROVED** [review]
Thank you for bringing this data online so quickly! Be sure to check [the results repo](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/az/arcgis) where you will see it show up in a few hours... and the new locations you've made accessible here will be live on vaccinateTheStates.com within the coming days!

I added one commit to fix the fetch-from URL - I should have shared the URL in the issue... otherwise you had no way to know what it should be, sorry about that!

**Pull request merged**